### PR TITLE
chore(tests): Add end-to-end testing system for query service

### PR DESCRIPTION
**Pull request by ahsanbarkati**
Add tests for cold storage and list disks

### CODE REVIEW

File: pkg/query-service/tests/test-deploy/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      - ./alertmanager.yml:/prometheus/alertmanager.yml
+    command:
+      - '--config.file=/prometheus/alertmanager.yml'
+      - '--storage.path=/data'
+
+# Notes for Maintainers/Contributors who will change Line Numbers of Frontend & Query-Section. Please Update Line Numbers in `./scripts/commentLinesForSetup.sh` & `./CONTRIBUTING.md`
+
+
+  query-service:
+    image: golang:1.17.8
```
Comment (prashant-shahi) [inline pkg/query-service/tests/test-deploy/docker-compose.yaml:27]: > golang:1.17.8

Replace it with `query-service:latest` as `make build-query-service-amd64` builds with `latest` tag by default unless overridden with `DOCKER_TAG`  env.

File: pkg/query-service/tests/docker.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	minio "github.com/minio/minio-go/v6"
+)
+
+const (
+	composeFile   = "./test-deploy/docker-compose.yaml"
```
Comment (prashant-shahi) [inline pkg/query-service/tests/docker.go:16]: > composeFile   = "./test-deploy/docker-compose.yaml"

Using `runtime.GOARCH`, we can conditionally switch between that that and `docker-compose.arm.yaml`.

File: pkg/query-service/tests/test-deploy/docker-compose.yaml (hunk 1 above)
Comment (ahsanbarkati) [reply pkg/query-service/tests/test-deploy/docker-compose.yaml]: Done

File: pkg/query-service/tests/cold_storage_test.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+	defer resp.Body.Close()
+	b, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.JSONEq(t, `[{"name":"default","type":"local"}, {"name":"s3","type":"s3"}]`, string(b))
+}
+
+func TestSetTTL(t *testing.T) {
+
+	testCases := []struct {
```
[the remark this answers is not in the document]
Comment (ahsanbarkati) [reply pkg/query-service/tests/cold_storage_test.go:52]: Done.

File: pkg/query-service/tests/docker.go (hunk 2 above)
Comment (ahsanbarkati) [reply pkg/query-service/tests/docker.go]: Done

**Review by ankitnayan — APPROVED** [review]

**Review by ankitnayan — APPROVED** [review]

**Pull request merged**